arbnode/transaction_streamer.go

```diff
@@ -1354,10 +1377,10 @@ func (s *TransactionStreamer) pollSubmittedTransactionForFinality(ctx context.Co
 	defer s.espressoTxnsStateInsertionMutex.Unlock()
 
 	batch := s.db.NewBatch()
-	if err := s.cleanEspressoSubmittedData(batch); err != nil {
-		return err
+	if err := s.setEspressoSubmittedTxns(batch, submittedTxns[0:]); err != nil {
```

Comment:
Should this be `[1:]` instead of `[0:]`? IIUC we have confirmed the first transaction in the list here is finalized so we want to retain only the subsequent transactions, deleting the first one

Reply:
89b22076